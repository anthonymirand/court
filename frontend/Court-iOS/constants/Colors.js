const tintColor = '#21ACA5';

export default {
  teal: '#21ACA5',
  tintColor,
  tabIconDefault: '#ccc',
  tabIconSelected: tintColor,
  tabBar: '#fefefe',
  errorBackground: 'red',
  errorText: '#fff',
  warningBackground: '#EAEB5E',
  warningText: '#666804',
  noticeBackground: tintColor,
  noticeText: '#fff',
  // To Do: remove above colors and dependencies
  lightPurple: '#9E90A2',
  coral: '#E3655B',
  mustard: '#E3B23C',
  wine: '#A4243B',
  pink: '#EA638C',
  nightSky: '#495867',
  lightBlue: '#8EB8E5',
  orange: '#EA8C55',
  peach: '#FFB997',
  darkBlue: '#28536B',
  purple: '#8783D1',
  mauve: '#BD9391',
  merlot: '#8B1E3F',
  blue: '#357DED',
  green: '#9DBF9E',
};
