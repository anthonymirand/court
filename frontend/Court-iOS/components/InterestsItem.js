import React from 'react';
import PropTypes from 'prop-types';
import { Icon, LinearGradient } from 'expo';

import {  Platform, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import Colors from '../constants/Colors';

import Avatar from '../components/Avatar';

/**
* Displays a singular interest with title, description, and image
*/
export default class InterestsItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = { selected: props.selected };
  }

  toggleSelected() {
    const { id, imgUrl, title, description, onAddInterest, onRemoveInterest } = this.props;
    const oldState = this.state;
    if (oldState.selected) {
      // Remove item
      onRemoveInterest(id);
    } else {
      // Add item
      onAddInterest(id, {title: title, description: description, imgUrl: imgUrl});
    }
    this.setState({ selected: !oldState.selected });
  }

  render() {
    const {
      id,
      imgUrl,
      disableSelect,
      showRemoveIcon,
      title,
      description,
      onRemoveInterest,
    } = this.props;

    const { selected } = this.state;

    return (
      <TouchableOpacity onPress={disableSelect ? null : this.toggleSelected.bind(this)} activeOpacity={0.75}>
        {selected ? (
          <LinearGradient style={showRemoveIcon ? styles.topSelectedWrapper : styles.selectedWrapper} colors={['#FA709A', '#FA709A']} start={[0, 0.5]} end={[1, 0.5]}>
            // Optional: add image preview here
            {imgUrl && (
              <Avatar width={40} src={imgUrl} />
            )}
            <View style={styles.textWrapper}>
              <Text style={styles.selectedTitleText}>
                {title}
              </Text>
              {description && (
                <Text style={styles.selectedDescriptionText}>
                  {description}
                </Text>
              )}
            </View>
            {showRemoveIcon && (
              // icon to remove interest item
              <View style={{marginLeft: 8, paddingTop: 1}}>
                <Icon.Ionicons
                  name='md-close'
                  size={25}
                  color='white'
                  onPress={() => onRemoveInterest(id)}
                />
              </View>
            )}
          </LinearGradient>
        ) : (
          <View style={styles.unselectedWrapper}>
            // Optional: add image preview here
            {imgUrl && (
              <Avatar width={40} src={imgUrl} />
            )}
            <View style={styles.textWrapper}>
              <Text style={styles.unselectedTitleText}>
                {title}
              </Text>
              {description && (
                <Text style={styles.unselectedDescriptionText}>
                  {description}
                </Text>
              )}
            </View>
            {showRemoveIcon && (
              // icon to remove interest item
              <View style={{marginLeft: 8, paddingTop: 1}}>
                <Icon.Ionicons
                  name='md-close'
                  size={25}
                  color='white'
                  onPress={() => onRemoveInterest(id)}
                />
              </View>
            )}
          </View>
        )}
      </TouchableOpacity>
    );
  }
}

InterestsItem.propTypes = {
  /**
  * Unique identifier for the given interest (from Google KGS)
  */
  id: PropTypes.string.isRequired,
  /**
  * URL for the interest's image preview
  */
  imgUrl: PropTypes.string,
  /**
  * Disables the onPress handler for pressing the interest
  */
  disableSelect: PropTypes.bool,
  /**
  * Displays an dismiss icon on the right of the component
  */
  showRemoveIcon: PropTypes.bool,
  /**
  * Title of the interest
  */
  title: PropTypes.string.isRequired,
  /**
  * Description of the interest
  */
  description: PropTypes.string,
  /**
  * Callback for removing a given interest (used in onPress handler for the removeIcon)
  */
  onRemoveInterest: PropTypes.func,
}

const styles = StyleSheet.create({
  unselectedWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderColor: Colors.peach,
    borderWidth: 1,
    borderRadius: 30,
    paddingVertical: 5,
    paddingHorizontal: 10,
    margin: 5,
  },
  topSelectedWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderColor: 'rgba(255, 255, 255, 0.0)',
    borderWidth: 1,
    borderRadius: 30,
    paddingVertical: 5,
    paddingHorizontal: 10,
    margin: 5,
  },
  selectedWrapper: {
    ...Platform.select({
      ios: {
        shadowColor: 'black',
        shadowOffset: { height: 8 },
        shadowOpacity: 0.15,
        shadowRadius: 9,
      },
      android: {
        elevation: 20,
      },
    }),
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderColor: 'rgba(255, 255, 255, 0.0)',
    borderWidth: 1,
    borderRadius: 30,
    paddingVertical: 5,
    paddingHorizontal: 10,
    margin: 5,
  },
  textWrapper: {
    flexDirection: 'column',
    alignItems: 'flex-start',
    marginLeft: 5,
  },
  unselectedTitleText: {
    fontSize: 20,
    color: Colors.peach,
  },
  selectedTitleText: {
    fontSize: 20,
    color: 'white',
  },
  unselectedDescriptionText: {
    fontSize: 15,
    color: 'rgba(247, 163, 138, 0.8)',
  },
  selectedDescriptionText: {
    fontSize: 15,
    color: 'white',
  }
});
