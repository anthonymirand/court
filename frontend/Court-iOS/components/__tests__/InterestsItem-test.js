import 'react-native';
import React from 'react';
import Colors from '../../constants/Colors';
import InterestsItem from '../InterestsItem';
import renderer from 'react-test-renderer';

it('Renders Default InterestsItem', () => {
  const tree = renderer.create(<InterestsItem id="kgs://12hg3f" title="The Office" description="American Sitcom" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Renders InterestsItem w/ Custom Color', () => {
  const tree = renderer.create(<InterestsItem id="kgs://12hg3f" title="The Office" description="American Sitcom" color={Colors.merlot}/>).toJSON();

  expect(tree).toMatchSnapshot();
});