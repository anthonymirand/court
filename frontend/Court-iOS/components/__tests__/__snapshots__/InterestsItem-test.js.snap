// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Default InterestsItem 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "white",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 5,
        "padding": 5,
      }
    }
  >
    // Optional: add image preview here
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "column",
          "marginLeft": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        The Office
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgba(255, 255, 255, 0.9)",
            "fontSize": 15,
          }
        }
      >
        American Sitcom
      </Text>
    </View>
  </View>
</View>
`;
